### DISCUSSION TITLE
How would I implement find_negative_cycle(s) ?

### DISCUSSION THREAD

**Discussion by Shane-Neeley** [networkx/networkx#5834] [Q&A]
There can be more than 1 negative cycle starting from a source node. Is there a way to find all of them?

https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.shortest_paths.weighted.find_negative_cycle.html#networkx.algorithms.shortest_paths.weighted.find_negative_cycle

**Comment by dschult** [comment]
If there is one, then there would typically be many. Are you restricting to simple cycles? Once way to explore how hard it is to eliminate negative cycles would be to find one -- then remove a node in that cycle from the graph and look for another. That ensures that the cycles stay simple. You could instead try removing the most negative edge in the cycle and looking again, but it depends what you are looking for.

What is the use case for this feature?  What are you trying to do with your graph? (what's the bigger question)

**Reply by Shane-Neeley** [reply]
Thanks. I'm trying to do a currency arbitrage application. Wondering if there were any standard ways to do this in networkx, or should I look for / build a custom solution.

I found this, "Bellman Ford, modified so that it returns cycles." --> https://github.com/robertmartin8/CryptoGraphArb/blob/d34ccdd433b207c693b0fe5cc0b7b4b60db423df/graph_arbitrage.py#L8
